lib/gems/pending/util/miq-password.rb

```diff
@@ -72,6 +73,40 @@ def self.encrypted?(str)
     !!split(str).first
   end
 
+  def self.encrypt_hash(values)
+    hash_values = values.deep_clone
+
+    hash_values.select { |f, _| f =~ /[Pp]assword/ }.each do |key, val|
+      next if val.strip.empty?
+
+      key_in_symbol = key.kind_of?(Symbol)
+      hash_values.delete(key)
+
+      key = "password::#{key}" unless key =~ /password::/
```

Comment:
I'm not actually a huge fan of this in general. `try_decrypt` already checks to see if the thing you gave it is encrypted so it doesn't feel like there should be a reason to use this to indicate that something is a password field. It feels like keeping the key regex's the same would be enough.